google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/Query.java

```diff
@@ -59,8 +61,64 @@ public static Query create(String tableId) {
     return new Query(tableId);
   }
 
-  private Query(String tableId) {
+  /**
+   * Constructs a new Query object with the specified authorized view id, which permits access to an
+   * explicit subset of the specified table. The authorized view id and table id will be combined
+   * with the instance name specified in the {@link
+   * com.google.cloud.bigtable.data.v2.BigtableDataSettings}.
+   *
+   * <p>See {@link com.google.cloud.bigtable.admin.v2.models.AuthorizedView} for more details about
+   * an AuthorizedView.
+   */
+  public static Query createForAuthorizedView(String tableId, String authorizedViewId) {
+    return new Query(tableId, authorizedViewId);
+  }
+
+  @InternalApi("For internal use only")
+  public static Query create(String tableId, @Nullable ExistsOptions existsOptions) {
```

Comment:
why do we having ExistsOptions, ReadRowOptions and ReadRowsOptions? for query I think we should only allow ReadRowsOptions? especially if it's internal use only, the caller should be able to call createForAuthorizedView instead?

Reply:
They are used in different places in BigtableDataClient.java, to convert a `ExistsOptions`/`ReadRowOptions`/`ReadRowsOptions` to a Query conveniently.

Reply:
Query should always use ReadRowsOptions. Can you point me to the code where other Options are used?

Reply:
The current `exists()` implementation actually builds a `Query` object:
https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java#L295-L314

Same thing for the `readRow()` method: https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java#L586-L592

That's why we convert those options into a `Query`, in existsAsync() and readRowAsync().

Reply:
Ah gotcha, I missed those. Thanks for clarifying!